tests/unit/mainwindow/statusbar/test_url.py

```diff
@@ -0,0 +1,126 @@
+                                    'warn', 'hover', 'normal'])
+
+
+@pytest.fixture
+def tab_widget():
+    """Fixture providing a fake tab widget."""
+    tab = namedtuple('Tab', 'cur_url load_status')
+    tab.load_status = namedtuple('load_status', 'name')
+    tab.cur_url = namedtuple('cur_url', 'toDisplayString')
+    return tab
+
+
+@pytest.fixture
+def url_widget(qtbot, monkeypatch, config_stub):
+    """Fixture providing a Url widget."""
+    config_stub.data = {
+        'colors': {
+            'statusbar.url.bg': 'white',
+            'statusbar.url.fg': 'black',
+            'statusbar.url.fg.success': 'yellow',
+            'statusbar.url.fg.success.https': 'green',
+            'statusbar.url.fg.error': 'red',
+            'statusbar.url.fg.warn': 'orange',
+            'statusbar.url.fg.hover': 'blue'},
+        'fonts': {},
+    }
+    monkeypatch.setattr(
+        'qutebrowser.mainwindow.statusbar.url.style.config', config_stub)
+    widget = UrlText()
+    qtbot.add_widget(widget)
+    assert not widget.isVisible()
+    return widget
+
+
+@pytest.mark.parametrize('url', [
+    ('http://abc123.com/this/awesome/url.html'),
+    ('https://supersecret.gov/nsa/files.txt'),
+    ('Th1$ i$ n0t @ n0rm@L uRL! P@n1c! <-->'),
+    (None)
+])
+def test_set_url(url_widget, url):
+    """Test text displayed by the widget."""
+    url_widget.set_url(url)
+    if url is not None:
+        assert url_widget.text() == url
+    else:
+        assert url_widget.text() == ""
+
+
+@pytest.mark.parametrize('url, title, text', [
+    ('http://abc123.com/this/awesome/url.html', 'Awesome site', 'click me!'),
+    ('https://supersecret.gov/nsa/files.txt', 'Secret area', None),
+    ('Th1$ i$ n0t @ n0rm@L uRL! P@n1c! <-->', 'Probably spam', 'definitely'),
+    (None, None, 'did I break?!')
+])
+def test_set_hover_url(url_widget, url, title, text):
+    """Test text when hovering over a link."""
+    url_widget.set_hover_url(url, title, text)
+    if url is not None:
+        assert url_widget.text() == url
+    else:
+        assert url_widget.text() == ''
+
+
+@pytest.mark.parametrize('status, expected', [
+    ('success', 'success'),
+    ('success_https', 'success_https'),
+    ('error', 'error'),
+    ('warn', 'warn')
+])
+def test_on_load_status_changed(url_widget, status, expected):
+    """Test text when status is changed."""
+    url_widget.set_url('www.example.com')
+    url_widget.on_load_status_changed(status)
+    assert url_widget._urltype.name == expected
+
+
+@pytest.mark.parametrize('load_status, url', [
+    ('success', 'http://abc123.com/this/awesome/url.html'),
+    ('success', 'http://reddit.com/r/linux'),
+    ('success_https', 'www.google.com'),
+    ('success_https', 'https://supersecret.gov/nsa/files.txt'),
+    ('warn', 'www.shadysite.org/some/path/to/a/file/that/has/issues.htm'),
+    ('error', 'Th1$ i$ n0t @ n0rm@L uRL! P@n1c! <-->'),
+    ('error', None)
```

Comment:
Similar to above, I'd prefer actual enum values here - you can then do `tab_widget.load_status = load_status` (and remove the `load_status` namedtuple from `tab_widget` as a nice side-effect), and `assert url_widget._urltype == load_status`.